com.visualstudio.code.insiders.yaml

```diff
@@ -0,0 +1,93 @@
+app-id: com.visualstudio.code.insiders
+default-branch: stable
+base: org.electronjs.Electron2.BaseApp
+base-version: '19.08'
+runtime: org.freedesktop.Sdk
+runtime-version: '19.08'
+sdk: org.freedesktop.Sdk
+command: code-insiders
+tags: [proprietary]
+desktop-file-name-suffix: ' (Flatpak)'
+separate-locales: false
+finish-args:
+  - --require-version=0.10.3
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --socket=ssh-auth
+  - --share=network
+  - --device=all
+  - --filesystem=host
+  - --allow=devel
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.freedesktop.secrets
+  - --talk-name=org.freedesktop.Flatpak
+  - --env=NPM_CONFIG_GLOBALCONFIG=/app/etc/npmrc
+  - --env=LD_LIBRARY_PATH=/app/lib
+  - --filesystem=xdg-config/kdeglobals:ro
+  - --talk-name=com.canonical.AppMenu.Registrar
+  - --talk-name=com.canonical.AppMenu.Registrar.*
+cleanup:
+  - /include
+  - /lib/pkgconfig
+  - /share/gtk-doc
+  - '*.la'
+modules:
+
+  - shared-modules/libsecret/libsecret.json
+
+  - name: vscode
+    buildsystem: simple
+    build-commands:
+      - install -D code-insiders.sh /app/bin/code-insiders
+      - install -Dm644 vscode_insiders_64.png /app/share/icons/hicolor/64x64/apps/com.visualstudio.code.insiders.png
+      - install -Dm644 vscode_insiders_128.png /app/share/icons/hicolor/128x128/apps/com.visualstudio.code.insiders.png
+      - install -Dm644 vscode_insiders_256.png /app/share/icons/hicolor/256x256/apps/com.visualstudio.code.insiders.png
+      - install -Dm644 vscode_insiders_512.png /app/share/icons/hicolor/512x512/apps/com.visualstudio.code.insiders.png
+      - install -Dm644 com.visualstudio.code.insiders.appdata.xml -t /app/share/metainfo
+      - install -Dm644 com.visualstudio.code.insiders.desktop -t /app/share/applications
+      - install -Dm644 npmrc -t /app/etc
+      - install -Dm644 flatpak-warning.txt -t /app/share/vscode
+      - install -D apply_extra -t /app/bin
+      - cp /usr/bin/ar /app/bin
+      - ARCH_TRIPLE=$(gcc --print-multiarch) && cp /usr/lib/${ARCH_TRIPLE}/libbfd-*.so
+        /app/lib
+      - ARCH_TRIPLE=$(gcc --print-multiarch) && ln -s /usr/lib/${ARCH_TRIPLE}/libtinfo.so/app/lib/libtinfo.so.5
+    sources:
+      - type: script
+        dest-filename: apply_extra
+        commands:
+          - ar x code-insiders.deb
+          - tar xf data.tar.xz
+          - mv usr/share/code-insiders vscode-insiders
+          - rm -r code-insiders.deb control.tar.gz data.tar.xz debian-binary usr
+      - type: file
+        path: code-insiders.sh
+      - type: file
+        path: flatpak-warning.txt
+      - type: file
+        path: npmrc
+      - type: file
+        path: com.visualstudio.code.insiders.appdata.xml
```

Comment:
New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html